src/schema/v2/artworksForUser/artworksForUser.ts

```diff
@@ -49,16 +49,20 @@ export const artworksForUser: GraphQLFieldConfig<void, ResolverContext> = {
       context
     )
 
-    const remainingSize = (gravityArgs.size || 0) - newForYouArtworks.length
     const backfillArtworks = await getBackfillArtworks(
-      remainingSize,
+      size || 0,
```

Comment:
I opted to request the maximum backfill items that could be necessary. Requesting _just enough_, and then excluding duplicates, could result in returning fewer than the requested number of results.

Also, I'm hoping passing the size through from the original query (rather than calculating an exact remaining size each time) leverages caching better.

Reply:
This makes sense to me.

Reply:
You should discuss with @nickskalkin b/c https://github.com/artsy/metaphysics/pull/5673/files just got merged which I believe undoes that in favor of un-cached _authenticated_ loaders (due to the need to filter out a user's disliked artworks)!